src/site/content/en/learn/pwa/service-workers/index.md

```diff
@@ -1,11 +1,137 @@
+   console.log("Service worker installed");
+});
+self.addEventListener("activate", event => {
+   console.log("Service worker activated");
+});
+```
+
+### Updating a service worker
+
+Service workers get updated when the browser detects that the service worker currently controlling the client and the live (from your server) version of the same file are byte-different.
+
+{% Aside 'warning' %}
+When updating your service worker, do so without renaming it, including adding file hashes to the filename. Otherwise, the browser will never get the new version of your service worker!
+{% endAside %}
+
+After a successful installation, the new service worker will wait to activate until the existing (old) service worker no longer controls any clients. This state is called "waiting", and it's how the browser ensures that only one version of your service worker is running at a time.
+Refreshing a page or reopening the PWA won't have the new service worker take effect. You need to close or navigate away from all tabs and windows using the current service worker and then navigate back for the new service worker to take control.
+Visit this [service worker lifecycle article](https://developers.google.com/web/fundamentals/primers/service-workers/lifecycle) for more information.
+
+## Service worker lifespan
+
+Once installed and registered, a service worker can manage all network requests within its scope. It runs on its  own thread, with activation and termination controlled by the browser. This lets it work even before or after your PWA is open! While service workers run on their own thread, there is no guarantee that in-memory state will persist between runs of a service worker, so make sure anything you want to reuse on each run is available either in IndexedDB or some other persistent storage.
+
+If not already running, a service worker will start whenever a network request in its scope is asked for, or when a triggering event, like periodic background sync or a push message, is received.
+
+Service workers don't live indefinitely. While exact timings differ between browsers, service workers will be terminated if they've been idle for a few seconds, or if they've been busy for too long. If a service worker has been terminated and an event occurs that would start it up, it will restart.
+
+## Capabilities
+
+With a registered and active service worker, you have a thread with a completely different execution lifecycle than the main one on your PWA. However, by default, the service worker file itself has no behavior. It won't cache or serve any resources, as this has to be done by your code. You'll find out how  in the following chapters.
+
+Service Worker's capabilities are not just for proxy or serving HTTP requests; other specs are available on top of it for other purposes, such as background code execution, web push notifications, and process payments. We'll discuss these additions in the [capabilities chapter](/learn/pwa/capabilities) chapter.
```

Comment:
```suggestion
Service Worker's capabilities are not just for proxy or serving HTTP requests; other features are available on top of it for other purposes, such as background code execution, web push notifications, and process payments. We'll discuss these additions in the [capabilities chapter](/learn/pwa/capabilities) chapter.
```

Reply:
I'm not not assuming that all competent web developers know anything about specs.